source_control/git.py

```diff
@@ -780,7 +804,10 @@ def main():
                     repo_updated = False
         if repo_updated is None:
             if module.check_mode:
-                module.exit_json(changed=True, before=before, after=remote_head)
+                result.update(changed=(result['before']!=remote_head), after=remote_head)
```

Comment:
This was required to ensure that check mode correctly returned a change compared to run mode. (Without this check-mode reported a change where run-mode did not)